lib/ember-qunit/qunit-module.js

```diff
@@ -45,25 +35,23 @@ export function createModule(Constructor, name, description, callbacks) {
   var module = new Constructor(name, description, callbacks);
 
   qunitModule(module.name, {
-    setup: function(assert) {
-      var done = assert.async();
-
+    beforeEach() {
       // provide the test context to the underlying module
       module.setContext(this);
 
-      return module.setup().then(function() {
+      return module.setup(...arguments).then(() => {
         if (beforeEach) {
-          beforeEach.call(module.context, assert);
+          beforeEach.apply(this, arguments);
         }
-      })['finally'](done);
+      });
     },
 
-    teardown: function(assert) {
+    afterEach() {
       if (afterEach) {
-        afterEach.call(module.context, assert);
+        afterEach.apply(this, arguments);
```

Comment:
I just noticed that a promise returned from the user's specified `afterEach` isn't handled properly (meaning we don't block on it before calling `module.teardown()` just below). We don't need to block this on fixing that, but it would be nice to get it fixed in a future refactor.